libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -325,6 +349,39 @@ class LC_CSR_CSC_Labeled_Graph
     return std::nullopt;
   }
 
+  /**
+   * Returns an edge iterator to an edge with some node and key with some label
+   * by searching for the key via the node's outgoing or incoming edges.
+   * If not found, returns nothing.
+   */
+  template <bool in_edges>
+  std::optional<edge_iterator> FindEdgeWithLabel(
+      GraphNode node, GraphNode key, const EdgeTy& data) const {
+    return FindFirstOrLastEdgeWithLabel<in_edges>(
+        false, false, node, key, data);
+  }
+
+  /**
+   * Returns an edge iterator to an edge with some node and key with some label
+   * by searching for the key via the node's outgoing or incoming edges.
+   * If not found, returns nothing.
+   */
+  template <bool in_edges>
+  std::optional<edges_iterator> FindEdgesWithLabel(
+      GraphNode node, GraphNode key, const EdgeTy& data) const {
+    auto first =
+        FindFirstOrLastEdgeWithLabel<in_edges>(true, false, node, key, data);
+    if (!first) {
+      return std::nullopt;
+    }
+
+    auto last =
+        FindFirstOrLastEdgeWithLabel<in_edges>(false, true, node, key, data);
```

Comment:
This may lead to some redundant work. I'd suggest changing this function to return a pair of first and last edges and have a flag saying all or any edge. That'll be faster.